service_config.go

```diff
@@ -164,7 +156,7 @@ type jsonMC struct {
 // TODO(lyuxuan): delete this struct after cleaning up old service config implementation.
 type jsonSC struct {
 	LoadBalancingPolicy *string
-	LoadBalancingConfig *internalserviceconfig.BalancerConfig
+	LoadBalancingConfig *json.RawMessage
```

Comment:
Why not continue to use the old logic of looping through list on the internalserviceconfig.BalancerConfig type? Now you need to prepare a graceful switch config alongside looping through this json list to find first config for balancer that is registered?

Reply:
> internalserviceconfig.BalancerConfig

I want to delete this type.  It's not a good idea to combine parsing and additional business logic in the same step.

Reply:
Yeah I remember our effort to do it. I had planned to do it as part of custom lb plumbing in xDS, but never got around to it. I think the configs are being validated twice IIRC as a result.